### PR TITLE
boards/samd21-based: model Kconfig

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This models Kconfig for all samd21-based boards. Currently this includes 28d84456e101229e3eb1926a4785aa0aebeca889  and c9b81f26550b50cb9cb06865126c1f0a2c9adca3 from #17299 as a dependency. Some minor changes on drivers are also included.

The list of boards is:
- adafruit-itsybitsy-m4
- arduino-mkr1000
- arduino-mkrfox1200
- arduino-mkrwan1300
- arduino-mkrzero
- arduino-nano-33-iot
- arduino-zero
- feather-m0
- feather-m0-lora
- feather-m0-wifi
- hamilton
- samd20-xpro
- samd21-xpro
- seeeduino_xiao
- sensebox_samd21
- serpente
- sodaq-autonomo
- sodaq-explorer
- sodaq-one
- sodaq-sara-aff
- sodaq-sara-sff
- wemos-zero

### Testing procedure
- Review the modelling and default configurations
- Green CI

### Issues/PRs references
Depends on #17299
Part of #16875

### CODE REVIEW

**Comment by fjmolinas** [comment]
For `hamilton` the feature conflict between `rtc` and `rtt` is activating I think the way out is forcing this in the application configuration, since which one is currently selected is only based on the order of inclusion.

**Comment by fjmolinas** [comment]
There also seems to be an issue with the `arduino` feature which is forcing `periph_uart` on make and not on Kconfig.

**Comment by fjmolinas** [comment]
T

> There also seems to be an issue with the `arduino` feature which is forcing `periph_uart` on make and not on Kconfig.

This is missing from Kconfig actually

**Comment by leandrolanzieri** [comment]
> T
> 
> > There also seems to be an issue with the `arduino` feature which is forcing `periph_uart` on make and not on Kconfig.
> 
> This is missing from Kconfig actually

Thanks, I fixed this now

**Comment by leandrolanzieri** [comment]
BTW: I rebased to resolve conflicts

**Comment by MrKevinWeiss** [comment]
ugg, tests/periph_pm still has some issues...

**Comment by fjmolinas** [comment]
This one now needs a rebase!

**Comment by fjmolinas** [comment]
Rebased and pushed a fix attempt

File: tests/periph_pm/Makefile.board.dep [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# For sam0 rtt conflicts with rtc, make will select only rtc because of
+# order of inclusion, but to resolve in the same way as Kconfig this also
+# selects ztimer_no_periph_rtt to select the same modules in Kconfig and make
+ifneq (,$(filter-out sam3,$(filter sam%,$(CPU))))
+  ifneq (,$(filter ztimer%,$(USEMODULE)))
+    USEMODULE += ztimer_no_periph_rtt
```
Comment (MrKevinWeiss) [inline tests/periph_pm/Makefile.board.dep:6]: It seems this works and it is commented well enough.  I feel like cpu specific code should not be in the application, but as this is somehow a workaround for the rather complex rtt/ztimer maybe we allow it for now until a better solution is found.

**Review by MrKevinWeiss — COMMENTED** [review]
It would be nice to have another maintainers opinion on the use of cpu  specific module selection in an application.

**Comment by MrKevinWeiss** [comment]
maybe ping @maribu or @kaspar030 ?

File: tests/periph_pm/Makefile.board.dep (hunk 1 above)
Comment (fjmolinas) [reply tests/periph_pm/Makefile.board.dep:6]: `Makefile.board.dep` is there exactly for this reason, sure most of the time its based on `$(BOARD)` but IMO this is exactly the kind of usage. So far we have force `Kconfig` to match `make` weird behaviour, I see no harm in doing it the other way around here. IMO we shouldn't stall this one!

Comment (MrKevinWeiss) [reply tests/periph_pm/Makefile.board.dep:6]: Fair enough.  It is documented and everything is passing.  Please squash!

**Comment by fjmolinas** [comment]
Rebase to fix conflict

**Review by MrKevinWeiss — APPROVED** [review]
ACK, if everything is documented and everything is green it is better to have the boards in then not.  This should prevent the whack-a-mole problem with the Kconfig migration.

**Review by fjmolinas — APPROVED** [review]
ACK! (all but my lately force-pushed changes)

**Comment by MrKevinWeiss** [comment]
Next will be #17402

**Pull request merged**